packages/elements-core/src/utils/http-spec/IServer.ts

```diff
@@ -57,3 +61,28 @@ export const getServerUrlWithDefaultValues = (server: IServer): string | null =>
 
   return stringifiedUrl.endsWith('/') ? stringifiedUrl.slice(0, -1) : stringifiedUrl;
 };
+
+export const getServerUrlWithFilledVariables = (server: IServer, values?: { [key: string]: string }): string | null => {
```

Comment:
You can probably get rid of the old getServerUrlWithDefaultValues function if we add this - calling this new one and only passing in the server should be identical to calling the old one, right?  

There are only a couple other files with references to that old function that need to be updated to call this one - build_request.ts and IServer.spec.ts.  

Should update those tests anyhow to add coverage for getServerUrlWithFilledVariables.